File: 3773-minimum-pair-removal-to-sort-array-i/minimum-pair-removal-to-sort-array-i.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var minimumPairRemoval = function(nums) {
    let count = 0;

    while (nums.length > 1) {
        let isSorted = true;
        let minSum = Infinity;
        let targetIndex = -1

        for (let i = 0; i < nums.length - 1; i++) {
            const sum = nums[i] + nums[i+1];

            if (nums[i] > nums[i+1]) {
                isSorted = false;
            }
            if (sum < minSum) {
                minSum = sum;
                targetIndex = i;
            }
        }

        if (isSorted) {
            break;
        }

        count++;

        nums[targetIndex] = minSum;
        nums.splice(targetIndex + 1, 1);
    }
    
    return count;
};

File: 3773-minimum-pair-removal-to-sort-array-i/minimum-pair-removal-to-sort-array-i.py
class Solution:
    def minimumPairRemoval(self, nums: List[int]) -> int:
        def isSorted(nums) -> bool:
            for i in range(1, len(nums)):
                if nums[i] < nums[i - 1]: return False
            return True
        result = 0

        while not isSorted(nums):
            min_sum = nums[0] + nums[1]
            index = 0

            for i in range(1, len(nums)-1):
                s = nums[i] + nums[i + 1]
                if s < min_sum:
                    min_sum = s
                    index = i

            nums[index] = min_sum
            nums.pop(index + 1)
            result += 1

        return result